### ISSUE TITLE
Case issue with ra-ui-materialui/esm/list/Filter.js

### ISSUE THREAD

**Issue by jpetitcolas** [marmelab/react-admin#5157]
We are facing a bug, occurring only on Mac OS (which is case insensitive):

> Failed to compile node_modules/ra-ui-materialui/esm/list/index.js:
> Cannot find file: 'filter.js' does not match the corresponding name on disk: 'node_modules/ra-ui-materialui/esm/list/Filter.js'.

And indeed, when checking the content of this folder, we have both a `Filter.js` file and a `filter` folder:

```txt
jpetitcolas@jpetitcolas-laptop:~/dev/tint/tint-platform$ ls -al node_modules/ra-ui-materialui/esm/list/ | grep -i filter
drwxrwxr-x  2 jpetitcolas jpetitcolas  4096 août  13 14:49 filter
-rw-r--r--  1 jpetitcolas jpetitcolas  3813 août  13 14:47 FilterButton.js
-rw-r--r--  1 jpetitcolas jpetitcolas  1026 août  13 14:47 FilterButtonMenuItem.js
-rw-r--r--  1 jpetitcolas jpetitcolas  2032 août  13 14:47 FilterFormInput.js
-rw-r--r--  1 jpetitcolas jpetitcolas  8033 août  13 14:47 FilterForm.js
-rw-r--r--  1 jpetitcolas jpetitcolas  2744 août  13 14:47 Filter.js
```

On a Linux workstation, there is no issue as it is case sensitive. But it prevents running the project on Mac. 

**Steps to reproduce:**

On a Mac, try to import:

```js
import { List } from 'react-admin';
```

**Other information:**

A workaround is to rename `Filter.js` file to `Filter_.js` file, and to change the import in the `node_modules/ra-ui-materialui/esm/list/index.js` file. 

**Environment**

* React-admin version: 3.8.1
* Browser: Chrome (Mac OS)

**Comment by djhi** [comment]
Thanks for reporting. It has been fixed in `3.8.2`

**Comment by fzaninotto** [comment]
Wait, you say you're having the problem on 3.8.1, not 3.8.0 ? The bug is supposed to be fixed in 3.8.1. 3.8.2 isn't out yet.

**Comment by jpetitcolas** [comment]
My bad! We are still on `3.8.0`. Sorry for the noise! 

TIL that `yarn info` retrieve information about the **last** version, not the one locally installed. Hence the issue. :)

**Comment by rgjr** [comment]
Confirmed fixed on 3.8.1, just updated after seeing the same error!
